router/throttler/factory.go

```diff
@@ -0,0 +1,171 @@
+package throttler
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/go-redis/redis/v8"
+
+	"github.com/rudderlabs/rudder-go-kit/config"
+	"github.com/rudderlabs/rudder-go-kit/stats"
+	"github.com/rudderlabs/rudder-go-kit/throttling"
+)
+
+type Factory interface {
+	Get(destName, destID string) Throttler
+	Shutdown()
+}
+
+// NewFactory constructs a new Throttler Factory
+func NewFactory(config *config.Config, stats stats.Stats) (Factory, error) {
+	f := &factory{
+		config:     config,
+		Stats:      stats,
+		throttlers: make(map[string]Throttler),
+	}
+	if err := f.initThrottlerFactory(); err != nil {
+		return nil, err
+	}
+	return f, nil
+}
+
+type factory struct {
+	config          *config.Config
+	Stats           stats.Stats
+	limiter         limiter
+	adaptiveLimiter limiter
+	throttlers      map[string]Throttler // map key is the destinationID
+	throttlersMu    sync.Mutex
+}
+
+func (f *factory) Get(destName, destID string) Throttler {
+	f.throttlersMu.Lock()
+	defer f.throttlersMu.Unlock()
+	defer func() {
+		if f.Stats != nil {
+			tags := stats.Tags{
+				"destination_id":   destID,
+				"destination_name": destName,
```

Comment:
I believe these are the labels we are using
```suggestion
				"destinationId":   destID,
				"destType": destName,
```

Reply:
Done